src/io/flutter/FlutterUtils.java

```diff
@@ -552,4 +555,10 @@ private static ModuleSourceOrderEntry findModuleSourceEntry(@NotNull Module modu
     }
     return null;
   }
+
+  public static boolean isPluginVersionDev() {
```

Comment:
I like it!
Should we also make sure that local builds behave like dev or provide an option to trigger dev/non-dev behavior?

Reply:
I could make this return true for `SNAPSHOT` versions as well. I'm not sure what an option would look like and whether we foresee adding more differentiated handling for dev vs. non-dev releases; I was thinking this current use case is very temporary and it's sufficient to test by changing plugin.xml to have a version name like `SNAPSHOT-dev`. Do you want to describe a potential option more?

Reply:
Making it return true for SNAPSHOT sounds good. As I'd like anyone building locally to also catch bugs before we enable features in STABLE.
The uses should be restricted to enabling certain experimental features by default instead of disabling them. Given that, maybe make this private and only in the FlutterSettings class to avoid misuse. It would be good to avoid uses of it that make it harder to test behavior in the release version by having dev/SNAPSHOT diverge.

Reply:
Okay, that makes sense - updated here: https://github.com/flutter/flutter-intellij/pull/4794